f5_openstack_agent/lbaasv2/drivers/bigip/pool_service.py

```diff
@@ -185,3 +185,35 @@ def _get_monitor_helper(self, service):
         else:
             hm = self.http_mon_helper
         return hm
+
+    def get_stats(self, service, bigips, stats):
+        """Return stat values for a single pool.
+
+        Stats to collect are defined as an array of strings in input stats.
+        Values are summed across one or more BIG-IPs defined in input bigips.
+
+        :param service: Has pool name/partition
+        :param bigips: One or more BIG-IPs to get pool stats from.
+        :param stats: Array of strings that define which stats to collect.
+        :return: A dict with key/value pairs for each stat defined in
+        input stats.
+        """
+        collected_stats = {}
+        for stat in stats:
+            collected_stats[stat] = 0
+
+        pool = self.service_adapter.get_pool(service)
+        part = pool["partition"]
+        for bigip in bigips:
```

Comment:
Can a pool span multiple BIGIPs?  Maybe check if the pool exists on the bigip before calling get_stats.